src/main/java/de/eposcat/master/approachImpl/EAV_DatabaseAdapter.java

```diff
@@ -6,150 +6,170 @@
     private static final String CREATE_ATTRIBUTE_VALUE_QUERY = "INSERT INTO eav_values(ent_id, att_id, value) VALUES (?,?,?)";
-    private static final String FIND_ATTRIBUTE_QUERY = "SELECT att_id FROM attributes WHERE datatype = ? AND name = ?";
+    private static final String FIND_ATTRIBUTE_QUERY = "SELECT id FROM attributes WHERE datatype = ? AND name = ?";
     private static final String CREATE_ATTRIBUTE_QUERY = "INSERT INTO attributes (datatype, name) VALUES (?,?)";
-    private static final String FIND_PAGE_BY_ID_QUERY = "SELECT * FROM entities WHERE entity_id = ?";
-    private static final String GET_PAGE_ATTRIBUTES_QUERY = "SELECT eav_values.att_id, value, datatype, name FROM eav_values INNER JOIN attributes ON eav_values.att_id = attributes.att_id WHERE ent_id = ?";
-    private static final String GET_CURRENT_ATTRIBUTE_IDS_QUERY = "SELECT att_id FROM eav_values WHERE ent_id = ?";
-    private static final String FIND_PAGE_BY_ATTRIBUTE_VALUE_QUERY = "SELECT eav_values.ent_id FROM attributes Inner JOIN eav_values ON eav_values.att_id = attributes.att_id WHERE attributes.name = ? AND eav_values.value = ?";
-    private static final String FIND_PAGE_BY_ATTRIBUTE_QUERY = "SELECT eav_values.ent_id FROM attributes Inner JOIN eav_values ON eav_values.att_id = attributes.att_id WHERE attributes.name = ?";
-    
-    private Connection conn;
-    private final static String ENTITYTABLE =  "entityTable";
-    private final static String ATTRIIBUTETABLE = "attributeTable";
-    
+    private static final String FIND_PAGE_BY_ID_QUERY = "SELECT * FROM entities WHERE id = ?";
+    private static final String FIND_PAGE_BY_TYPE_QUERY = "SELECT id FROM entities WHERE typename = ?";
+    private static final String GET_PAGE_ATTRIBUTES_QUERY = "SELECT eav_values.att_id, value, datatype, name FROM eav_values INNER JOIN attributes ON eav_values.att_id = attributes.id WHERE ent_id = ?";
+    private static final String GET_LAST_ATTRIBUTE_IDS_QUERY = "SELECT att_id FROM eav_values WHERE ent_id = ?";
+    private static final String FIND_PAGE_BY_ATTRIBUTE_VALUE_QUERY = "SELECT eav_values.ent_id FROM attributes Inner JOIN eav_values ON eav_values.att_id = attributes.id WHERE attributes.name = ? AND eav_values.value = ?";
+    private static final String FIND_PAGE_BY_ATTRIBUTE_QUERY = "SELECT eav_values.ent_id FROM attributes Inner JOIN eav_values ON eav_values.att_id = attributes.id WHERE attributes.name = ?";
+
+    private final Connection conn;
+    private final static String ENTITY_TABLE =  "entityTable";
+    private final static String ATTRIBUTE_TABLE = "attributeTable";
+
     public EAV_DatabaseAdapter(AbstractConnectionManager connectionManager) {
         this.conn = connectionManager.getConnection();
     }
 
     @Override
-    public Page createPage(String typeName) throws SQLException{
+    public Page createPage(String typeName) throws SQLException {
```

Comment:
```suggestion
    @Nonnull
    public Page createPage(String typeName) throws SQLException {
```
If this method always returns a non-null object, you could annotate it with @Nonnull (javax.annotation.Nonnull)

Reply:
Is there a certain dependency I need to use to get @Nonnull?

My IDE does not find the annotation at the moment, only @NotNull.